src/shared/components/getting-started-section/index.js

```diff
@@ -1,30 +1,157 @@
+  node.files.cat('${cid}', (err, data) => {
+    if (err) return console.error(err)
+
+    // convert Buffer back to string
+    console.log(data.toString())
+  })
+})`
+
+class GettingStarted extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {ipfsLoaded: false, outputAdd: null, outputGet: null, cid: null}
+    this.scopeAdd = null
+    this.scopeGet = null
+    this.refAdd = React.createRef()
+    this.refGet = React.createRef()
+    this.ipfs = null
+  }
+
+  handleChange = (editor) => (content) => {
+    if (content instanceof Error) {
+      content = content.message
+    }
+
+    if (typeof content !== 'string') {
+      return
+    }
+
+    if (editor === 'add') {
+      this.setState({outputAdd: content})
+    } else {
+      this.setState({outputGet: content})
+    }
+
+    if (content) {
+      if (typeof content === 'string' && content.length === 46 && content !== this.state.cid) {
+        preload(content)
+        this.setState({cid: content})
+      }
+    }
+  }
+  handleRunClick = (editor) => () => {
+    if (editor === 'add') {
+      this.refAdd.current.run()
+    } else {
+      this.refGet.current.run()
+    }
+  }
+
+  componentDidMount () {
+    getIpfs().then(stubed => {
+      this.ipfs = stubed
+      this.scopeAdd = {IPFS: this.ipfs, console: log(this.handleChange('add'))}
+      this.scopeGet = {IPFS: this.ipfs, console: log(this.handleChange('get'))}
+      this.setState({ipfsLoaded: true})
+    }).catch(console.error)
+  }
+
+  render () {
+    const { intl: { messages } } = this.props
+    return (
+      <div className={ styles.container }>
+        <div className={ styles.backgroundSvg }>
+          <div className={ styles.hex1 }><HexSvg /></div>
+          <div className={ styles.hex2 }><HexSvg /></div>
+        </div>
+        <div className={ styles.content }>
+          <h1>{ messages.gettingStarted.sectionTitle }</h1>
+          <span className={ styles.sectionDescription }>
+            <p>{ messages.gettingStarted.sectionDesc }</p>
+          </span>
+          <div className={ styles.panel } >
+            <p className={ styles['LiveSnippet-title'] }>{messages.gettingStarted.addDataToIPFS}</p>
+            <LiveProvider key="add" className={styles.LiveSnippet} code={code} scope={this.scopeAdd} mountStylesheet={false} transformCode={transformCode}>
+              <div style={{ position: 'relative' }}>
+                <LiveEditor ref={this.refAdd} className={'language-js ' + styles['LiveSnippet-editor']}/>
+                <button className={styles['LiveSnippet-run']} onClick={this.handleRunClick('add')}>Run</button>
+              </div>
+              <div className={styles['LiveSnippet-preview']} >
+                <p className={styles['LiveSnippet-output']}>{messages.gettingStarted.output}</p>
+                <pre>
+                  <code>{this.state.outputAdd}</code>
+                </pre>
+                {this.state.ipfsLoaded && <LivePreview/>}
```

Comment:
The `ipfsLoader` is being used in several places. Declare all the needed `state` properties before return